e2e/playwright/tests/folder-rules/page-objects/components/actions-dropdown.component.ts

```diff
@@ -10,23 +10,84 @@ import { Locator, Page } from '@playwright/test';
 }
 
 export class ActionsDropdownComponent extends BaseComponent {
-  private static rootElement = '.mat-select-panel';
+  private static rootElement = 'aca-rule-action-list';
 
-  public getOptionLocator = (optionName: string): Locator => this.getChild('.mat-option-text', { hasText: optionName });
+  public getOptionLocator = (optionName: string): Locator => this.page.locator('.mat-select-panel .mat-option-text', { hasText: optionName });
+  private ruleActionLocator = this.getChild('aca-rule-action [data-automation-id="rule-action-card-view"]');
+  private addActionButtonLocator = this.getChild('[data-automation-id="rule-action-list-add-action-button"]');
+  private actionDropdownLocator = this.getChild('[data-automation-id="rule-action-select"]');
+  private actionAspectNameLocatorText = '[data-automation-id="header-aspect-name"] mat-select';
+  private actionCheckInInputLocatorText = '[data-automation-id="header-description"] input';
+  private actionAutoDeclareLocatorText = '[data-automation-id="header-version"] mat-select';
+  private actionSimpleWorkflowStepInputLocatorText = '[data-automation-id="header-approve-step"] input';
+  private actionSimpleWorkflowApproveFolderText = `[data-automation-id="header-approve-folder"] input`;
+  private actionSimpleWorkflowActionChoiceText = '[data-automation-id="content-node-selector-actions-choose"]';
+  private actionSimpleWorkflowLabelApproveLocatorText = `[data-automation-id="card-boolean-label-approve-move"]`;
+  private actionSimpleWorkflowSRejectStepLocatorText = '[data-automation-id="header-reject-step"] input';
+  private actionSimpleWorkflowRejectFolderLocatorText = `[data-automation-id="header-reject-folder"] input`;
 
   constructor(page: Page) {
     super(page, ActionsDropdownComponent.rootElement);
   }
 
-  async selectAction(action: Partial<ActionType>): Promise<void> {
-    await this.page.locator(`aca-edit-rule-dialog [data-automation-id="rule-action-select"]`).click();
+  async selectAction(action: Partial<ActionType>, index: number): Promise<void> {
+    if (index > 0) {
+      await this.addActionButtonLocator.click();
+    }
+    await this.actionDropdownLocator.nth(index).click();
     const option = this.getOptionLocator(action);
-    await option.click();
+    await option.nth(0).click();
+  }
+
+  async dropdownSelection(selectValue: string, locator: string, index: number): Promise<void> {
+    await this.ruleActionLocator.nth(index).locator(locator).click();
+    const option = this.getOptionLocator(selectValue);
+    await option.nth(0).click();
+  }
+
+  async insertCheckInActionValues(checkInValue: string, index: number): Promise<void> {
+    await this.ruleActionLocator.nth(index).locator(this.actionCheckInInputLocatorText).type(checkInValue, { delay: 50 });
+  }
+
+  async insertAddAspectActionValues(AspectValue: string, index: number): Promise<void> {
+    await this.dropdownSelection(AspectValue, this.actionAspectNameLocatorText, index);
```

Comment:
fyi, for this type of functions, you don't need to use `await` and can just `return` the function, so that the calling code can await it...